src/model.ts

```diff
@@ -1,13 +1,38 @@
+require('isomorphic-fetch');
+import * as _ from 'lodash';
 import { Request } from 'express';
-import { Readable } from 'stream';
+import { PagingStream } from './paging-stream';
+import { searchContent } from '@esri/hub-search';
+import { IContentSearchRequest } from './types';
 
 export class HubApiModel {
-  async getStream (request: Request) {
-    console.log(request);
 
-    const ret = new Readable({ read () {} });
-    ret.push({ foo: 'bar' });
-    ret.push(null);
-    return ret;
+  getStream (request: Request) {
+    const searchRequest: IContentSearchRequest = request.res.locals.searchRequest;
+
+    if (!_.has(searchRequest, 'options.portal')) {
+      _.set(searchRequest, 'options.portal', 'https://www.arcgis.com');
+    }
+
+    const searchApiStream = new PagingStream({
+      firstPageParams: searchRequest,
+
+      loadPage: (params: IContentSearchRequest | any) => {
```

Comment:
Should this be `async`?

Reply:
I don't think so since we don't have to use `await` in the body.